.github/workflows/ci.yml

```diff
@@ -32,6 +32,9 @@ jobs:
       with:
         fetch-depth: 0
 
+    - name: numpy>=1.21.0 for type checkers
```

Comment:
hmm this is very fragile if we change this

Reply:
Adding on to this. Can we just make two conda envs in the CI, one for typing and one for everything else?

Reply:
Currently, we don't need this to force numpy>1.21.0 because pip/conda already pick numpy 1.21.*.

We could simply remove this change. If at some point the CI picks a too old numpy, we will know very quickly.

Reply:
It is probably best to decouple the CI-change from the pre-commit change. I will remove the CI-changes from this PR.

Reply:
if you saw #43804 numpy gets installed <1.21 since that is a numba requirement when resolving dependencies.

Reply:
> if you saw #43804 numpy gets installed <1.21 since that is a numba requirement when resolving dependencies.

Yes, that is why I tried to add this line, but it seems the CI prefers an older numba in favor of a newer numpy :)

Reply:
Just to be clear, the goal of this PR has changed from "having numpy>1.21. on the CI and for pre-commit" to "not having pyright as a mandatory part of pre-commit".